pkg/account/deployer/client.go

```diff
@@ -0,0 +1,268 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package deployer
+
+import (
+	"context"
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/clients/accounts"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+	accountmanagement "github.com/dynatrace/dynatrace-configuration-as-code-core/gen/account_management"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"io"
+	"net/http"
+	"slices"
+)
+
+type (
+	Permissions    = accountmanagement.PermissionsDto
+	Policy         = accountmanagement.CreateOrUpdateLevelPolicyRequestDto
+	Group          = accountmanagement.PutGroupDto
+	ManagementZone = accountmanagement.ManagementZoneResourceDto
+
+	accountManagementClient struct {
+		accountInfo          AccountInfo
+		supportedPermissions []remoteId
+		client               *accounts.Client
+	}
+)
+
+func NewClient(info AccountInfo, client *accounts.Client, supportedPermissions []remoteId) *accountManagementClient {
+	return &accountManagementClient{
+		accountInfo:          info,
+		client:               client,
+		supportedPermissions: supportedPermissions,
+	}
+}
+
+func (d *accountManagementClient) getAccountInfo() AccountInfo {
+	return d.accountInfo
+}
+
+func (d *accountManagementClient) getGlobalPolicies(ctx context.Context) (map[string]remoteId, error) {
+	globalPolicies, _, err := d.client.PolicyManagementAPI.GetLevelPolicies(ctx, "global", "global").Execute()
```

Comment:
is there no need for the full error check that looks at the error as well as the response here?